diesel/src/sqlite/connection/raw.rs

```diff
@@ -379,6 +400,40 @@ unsafe fn null_aggregate_context_error(ctx: *mut ffi::sqlite3_context) {
     );
 }
 
+#[allow(warnings)]
+extern "C" fn run_collation_function<F>(
+    user_ptr: *mut libc::c_void,
+    len_rhs: libc::c_int,
+    rhs_ptr: *const libc::c_void,
+    len_lhs: libc::c_int,
+    lhs_ptr: *const libc::c_void,
+) -> libc::c_int
+where
+    F: FnMut(&str, &str) -> std::cmp::Ordering + Send + 'static,
+{
+    unsafe {
+        let user_ptr = user_ptr as *mut F;
+        let f = match user_ptr.as_mut() {
+            Some(f) => f,
+            None => {
+                //FIXME
+                return 0;
+            }
+        };
+
+        //FIXME: check args
+        let rhs =
```

Comment:
We should have a comment here indicating that the provided strings are `utf8` because we specified that above and otherwise that would be a bug in sqlite.